app/code/Magento/Catalog/Test/Unit/Model/ProductRepositoryTest.php

```diff
@@ -313,23 +313,37 @@ public function testGetProductInEditMode()
             ->will($this->returnValue('test_id'));
         $this->productMock->expects($this->once())->method('setData')->with('_edit_mode', true);
         $this->productMock->expects($this->once())->method('load')->with('test_id');
-        $this->productMock->expects($this->once())->method('getSku')->willReturn($sku);
+        $this->productMock->expects($this->once())->method('getId')->willReturn('test_id');
         $this->assertEquals($this->productMock, $this->model->get($sku, true));
     }
 
     public function testGetBySkuWithSpace()
```

Comment:
This test does not seem to test initial case anymore. Please change so that it actually checks trimmed SKU.